src/Npgsql/NpgsqlConnector.cs

```diff
@@ -536,16 +533,18 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
                         break;
                     case 'S':
                         var clientCertificates = new X509CertificateCollection();
-                        if (Settings.ClientCertificate != null)
-                            clientCertificates.Add(new X509Certificate(Settings.ClientCertificate));
-                        else if (Environment.GetEnvironmentVariable("PGSSLCERT") is string envCertPath)
-                            clientCertificates.Add(new X509Certificate(envCertPath));
-                        else if (Environment.GetEnvironmentVariable(PGUtil.IsWindows ? "APPDATA" : "HOME") is string homeDir)
+                        var certPath = Settings.ClientCertificate ?? PostgresEnvironment.SslCert;
+                        var certPathExists = true;
+
+                        if (certPath is null)
                         {
-                            var certPath = Path.Combine(homeDir, "postgresql", "postgresql.crt");
-                            if (File.Exists(certPath))
-                                clientCertificates.Add(new X509Certificate(certPath));
+                            certPath = PostgresEnvironment.SslCertDefault;
+                            certPathExists = File.Exists(certPath);
```

Comment:
Can we match the [pattern](https://github.com/npgsql/npgsql/pull/2657/files?file-filters%5B%5D=.cs#diff-d4be05c29dce27ce00fbf520f6b09248R370-R371) used for `passFilePath`?

Reply:
What do you mean?

Reply:
E.g.

```c#
var passFilePath = Settings.Passfile ?? PostgresEnvironment.PassFile ?? PostgresEnvironment.PassFileDefault;
if (passFilePath != null && File.Exists(passFilePath))
...
```
```c#
var certPath = Settings.ClientCertificate ?? PostgresEnvironment.SslCert ?? PostgresEnvironment.SslCertDefault;
if (certPath != null && File.Exists(certPath))
...
```

Reply:
No, there was a discussion which is lost (probably because of rebases, but this shouldn't happen as I know) that we should throw if file is specified explicitly, but it doesn't exist. Makes sense to apply the same behavior to `Passfile`.

Reply:
Opened #2694 to track the issue.